source/DSCResources/DSC_CMForestDiscovery/DSC_CMForestDiscovery.psm1

```diff
@@ -0,0 +1,299 @@
+
+    .PARAMETER Enabled
+        Specifies the enablement of the forest discovery method. If settings is set to $false no other value provided will be
+        evaluated for compliance.
+
+    .PARAMETER PollingSchedule
+        Specifies a schedule and determines how often Configuration Manager attempts to discover Active Directory forest.
+
+    .PARAMETER EnableActiveDirectorySiteBoundaryCreation
+        Indicates whether Configuration Manager creates Active Directory boundaries from AD DS discovery information.
+
+    .PARAMETER EnableSubnetBoundaryCreation
+        Indicates whether Configuration Manager creates IP address range boundaries from AD DS discovery information.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $PollingSchedule,
+
+        [Parameter()]
+        [Boolean]
+        $EnableActiveDirectorySiteBoundaryCreation,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSubnetBoundaryCreation
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+        if ($Enabled -eq $true)
+        {
+            $exludelist = @('Verbose','PollingSchedule')
+            foreach ($param in $PSBoundParameters.GetEnumerator())
+            {
+                if ($exludelist -notcontains $param.key)
+                {
+                    if ($param.Value -ne $state[$param.key])
+                    {
+                        Write-Verbose -Message ($script:localizedData.SettingSettings -f $param.Key, $param.Value)
+                        $buildingParams += @{
+                            $param.Key = $param.Value
+                        }
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($PollingSchedule))
+            {
+                $newSchedule = @{
+                    RecurInterval = $PollingSchedule.RecurInterval
+                    RecurCount    = $PollingSchedule.RecurCount
+                }
+
+                $desiredPollingSchedule = New-CMSchedule @newSchedule
+
+                $currentSchedule = @{
+                    RecurInterval = $state.PollingSchedule.RecurInterval
+                    RecurCount    = $state.PollingSchedule.RecurCount
+                }
+
+                $stateSchedule = New-CMSchedule @currentSchedule
+
+                $array = @('DayDuration','DaySpan','HourDuration','HourSpan','IsGMT','MinuteDuration','MinuteSpan')
+
+                foreach ($item in $array)
+                {
+                    if (($desiredPollingSchedule).$($item) -ne ($stateSchedule).$($item))
```

Comment:
This: if (($desiredPollingSchedule).$($item) -ne ($stateSchedule).$($item))
Could be: if ($desiredPollingSchedule.$($item) -ne $stateSchedule.$($item))

Reply:
Changed in both test and set.